lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -1514,6 +1514,7 @@ void replaceUsesAndPropagateType(OpBuilder &builder, Operation *oldUse,
   // Perform late replacement.
   for (OpOperand *operand : operandsToReplace) {
     if (auto wait = dyn_cast<ttng::WarpGroupDotWaitOp>(operand->getOwner())) {
+      OpBuilder::InsertionGuard guard(builder);
```

Comment:
nit: I would move it to the top of the function. This usually how it is done to make sure the insert point gets restored for the caller